### EDA GAMING Video_Games_Sales

### Campos:
- Name
    - Nombre del VideoJuego
- Platform
    - Plataforma
- Year_of_Release
    - Año de Lanzamiento
- Genre
    - Genero
- Publisher
    - Casa que Publica
- NA_Sales
    - Ventas en Norte America
- EU_Sales
    - Ventas en Union Europea
- JP_Sales
    - Ventas en Japon
- Other_Sales
    - Ventas en otros Lugares
- Global_Sales
    - Ventas Globales
- Critic_Score
    - Calificación de la Critica
- Critic_Count
    - Cantidad de Calificaciones
- User_Score
    - Puntuacion Usuario
- User_Count
    - Recuento de Usuarios
- Developer
    - Desarrollador
- Rating
   - Clasificación

In [ ]:
# %config InlineBackend.figure_format = 'retina'
# pd.set_option('display.max_columns', None)

Importando Librerias

In [1]:
import polars as pl
import duckdb

In [2]:
df = pl.read_csv('csv_originales/04_Video_Games_Sales.csv',
                 separator=',',
                 null_values=["N/A", "NA"])

In [3]:
df

shape: (16_719, 16)
┌────────────┬──────────┬────────────┬────────────┬───┬───────────┬───────────┬───────────┬────────┐
│ Name       ┆ Platform ┆ Year_of_Re ┆ Genre      ┆ … ┆ User_Scor ┆ User_Coun ┆ Developer ┆ Rating │
│ ---        ┆ ---      ┆ lease      ┆ ---        ┆   ┆ e         ┆ t         ┆ ---       ┆ ---    │
│ str        ┆ str      ┆ ---        ┆ str        ┆   ┆ ---       ┆ ---       ┆ str       ┆ str    │
│            ┆          ┆ i64        ┆            ┆   ┆ str       ┆ i64       ┆           ┆        │
╞════════════╪══════════╪════════════╪════════════╪═══╪═══════════╪═══════════╪═══════════╪════════╡
│ Wii Sports ┆ Wii      ┆ 2006       ┆ Sports     ┆ … ┆ 8         ┆ 322       ┆ Nintendo  ┆ E      │
│ Super      ┆ NES      ┆ 1985       ┆ Platform   ┆ … ┆           ┆ null      ┆           ┆        │
│ Mario      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Bros.      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Mario Kart ┆ Wii      ┆ 2008       ┆ Racing     ┆ … ┆ 8.3       ┆ 709       ┆ Nintendo  ┆ E      │
│ Wii        ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Wii Sports ┆ Wii      ┆ 2009       ┆ Sports     ┆ … ┆ 8         ┆ 192       ┆ Nintendo  ┆ E      │
│ Resort     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Pokemon    ┆ GB       ┆ 1996       ┆ Role-Playi ┆ … ┆           ┆ null      ┆           ┆        │
│ Red/Pokemo ┆          ┆            ┆ ng         ┆   ┆           ┆           ┆           ┆        │
│ n Blue     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ …          ┆ …        ┆ …          ┆ …          ┆ … ┆ …         ┆ …         ┆ …         ┆ …      │
│ Samurai    ┆ PS3      ┆ 2016       ┆ Action     ┆ … ┆           ┆ null      ┆           ┆        │
│ Warriors:  ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Sanada     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Maru       ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ LMA        ┆ X360     ┆ 2006       ┆ Sports     ┆ … ┆           ┆ null      ┆           ┆        │
│ Manager    ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ 2007       ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Haitaka no ┆ PSV      ┆ 2016       ┆ Adventure  ┆ … ┆           ┆ null      ┆           ┆        │
│ Psychedeli ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ ca         ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Spirits &  ┆ GBA      ┆ 2003       ┆ Platform   ┆ … ┆           ┆ null      ┆           ┆        │
│ Spells     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Winning    ┆ PSV      ┆ 2016       ┆ Simulation ┆ … ┆           ┆ null      ┆           ┆        │
│ Post 8     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ 2016       ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
└────────────┴──────────┴────────────┴────────────┴───┴───────────┴───────────┴───────────┴────────┘

Determinando los Distintos Generos en las consolas

In [8]:
generos = []
for genero in df['Genre'].unique():
    generos.append(genero)

print(generos)

['Role-Playing', 'Sports', 'Platform', 'Puzzle', 'Adventure', 'Strategy', 'Shooter', 'Racing', 'Fighting', 'Simulation', 'Misc', None, 'Action']


In [5]:
# df.group_by('Genre').count()   # Manera ya deprecada
df.group_by('Genre').len()

shape: (13, 2)
┌──────────────┬──────┐
│ Genre        ┆ len  │
│ ---          ┆ ---  │
│ str          ┆ u32  │
╞══════════════╪══════╡
│ Simulation   ┆ 874  │
│ Sports       ┆ 2348 │
│ Shooter      ┆ 1323 │
│              ┆ 2    │
│ Racing       ┆ 1249 │
│ …            ┆ …    │
│ Platform     ┆ 888  │
│ Strategy     ┆ 683  │
│ Misc         ┆ 1750 │
│ Role-Playing ┆ 1500 │
│ Adventure    ┆ 1303 │
└──────────────┴──────┘

Cuantos Publisher Hay?

In [11]:
df['Developer'].unique().count()

1696

#### Datos Faltantes

In [12]:
for column in df.columns:
    print(f"{column} -->  {df[column].is_null().sum()}")

Name -->  2
Platform -->  0
Year_of_Release -->  269
Genre -->  2
Publisher -->  54
NA_Sales -->  871
EU_Sales -->  1199
JP_Sales -->  0
Other_Sales -->  1378
Global_Sales -->  991
Critic_Score -->  8582
Critic_Count -->  8582
User_Score -->  9129
User_Count -->  9129
Developer -->  6623
Rating -->  6769


***
## Inputacion Datos Faltantes en Columnas Asociadas a Ventas

#### Prestar Atención a la Fórmula
`Global_Sales = Other_Sales + JP_Sales + EU_Sales + NA_Sales` <br>
`Global_Sales` ---> Columna 9   --->  Ventas GLOBALES<br>
`Other_Sales` ---> Colummna 8   --->  Ventas OTROS<br>
`JP_Sales` ---> Columna 7       --->  Ventas JAPON<br>
`EU_Sales` ---> Columna 6       --->  Ventas UNION EUROPEA<br>
`NA_Sales` ---> Columna 5       --->  Ventas NORTE AMERICA<br>
Recorrer TODOS los registros y aplicar la Fórmula donde se pueda<br>
`Ejemplo`<br>
Si en una fila `Other_Sales` es null o cero(0) y Todas las demas columnas tienen informacion numerica entonces<br>
`Other_Sales` = `Global_Sales` - `JP_Sales` - `EU_Sales` - `NA_Sales` <br>
y asi sucesivamente hay que estudiar todos los casos...


In [6]:
for i in range(len(df)):
    if df[i,5] is not None and\
       df[i,6] is not None and\
       df[i,7] is not None and\
       df[i,9] is not None and\
       (df[i,8] is None or df[i,8] == 0):
        df[i,8] = df[i,9] - df[i,7] - df[i,6] - df[i,5]
        continue
      
    if df[i,5] is not None and\
       df[i,6] is not None and\
       df[i,7] is not None and\
       df[i,8] is not None and\
       (df[i,9] is None or df[i,9] == 0):
        df[i,9] = df[i,8] + df[i,7] + df[i,6] + df[i,5]
        continue

    if df[i,6] is not None and\
       df[i,7] is not None and\
       df[i,8] is not None and\
       df[i,9] is not None and\
       (df[i,5] is None or df[i,5] == 0):
        df[i,5] = df[i,9] - df[i,8] - df[i,7] - df[i,6]
        continue

    if df[i,5] is not None and\
       df[i,7] is not None and\
       df[i,8] is not None and\
       df[i,9] is not None and\
       (df[i,6] is None or df[i,6] == 0):
        df[i,6] = df[i,9] - df[i,8] - df[i,7] - df[i,5]
        continue    

Con la Aplicación de la Formula en todos los registros Veamos como quedan los registros Faltantes Solo en columnas de Ventas

In [7]:
print(f"NA_Sales -> {df['NA_Sales'].is_null().sum()}   Antes Habían -->871")
print(f"EU_Sales -> {df['EU_Sales'].is_null().sum()}   Antes Habían -->1199")
print(f"JP_Sales -> {df['JP_Sales'].is_null().sum()}")
print(f"Other_Sales -> {df['Other_Sales'].is_null().sum()}   Antes Habían -->1378")
print(f"Global_Sales -> {df['Global_Sales'].is_null().sum()}   Antes Habían -->991")

NA_Sales -> 171   Antes Habían -->871
EU_Sales -> 214   Antes Habían -->1199
JP_Sales -> 0
Other_Sales -> 234   Antes Habían -->1378
Global_Sales -> 205   Antes Habían -->991


In [8]:
df[0]

shape: (1, 16)
┌────────────┬──────────┬──────────────┬────────┬───┬────────────┬────────────┬───────────┬────────┐
│ Name       ┆ Platform ┆ Year_of_Rele ┆ Genre  ┆ … ┆ User_Score ┆ User_Count ┆ Developer ┆ Rating │
│ ---        ┆ ---      ┆ ase          ┆ ---    ┆   ┆ ---        ┆ ---        ┆ ---       ┆ ---    │
│ str        ┆ str      ┆ ---          ┆ str    ┆   ┆ str        ┆ i64        ┆ str       ┆ str    │
│            ┆          ┆ i64          ┆        ┆   ┆            ┆            ┆           ┆        │
╞════════════╪══════════╪══════════════╪════════╪═══╪════════════╪════════════╪═══════════╪════════╡
│ Wii Sports ┆ Wii      ┆ 2006         ┆ Sports ┆ … ┆ 8          ┆ 322        ┆ Nintendo  ┆ E      │
└────────────┴──────────┴──────────────┴────────┴───┴────────────┴────────────┴───────────┴────────┘

Verificar cuantos son los registros TOTALES con al menos 2 Campos de Ventas NULOS. (Recordar que el aplicar la formula en uno de los pasos anteriores eliminó la posibilidad de haya un solo campo de venta NULO) <br>


In [9]:
con_ventas_nulas = df.filter((df['Global_Sales'].is_null()) |\
                             (df['EU_Sales'].is_null()) |\
                             (df['Other_Sales'].is_null()) |\
                             (df['NA_Sales'].is_null())
                            )

len(con_ventas_nulas)

401

In [10]:
sin_ventas_nulas = df.filter((~df['Global_Sales'].is_null()) &\
                             (~df['EU_Sales'].is_null()) &\
                             (~df['Other_Sales'].is_null()) &\
                             (~df['NA_Sales'].is_null())
                            )

len(sin_ventas_nulas)

16318

Hay 401 Registros con al menos 2 Zonas de Ventas Nulas de un total de 16719 Registros 2,4% de los Registros

In [21]:
sin_ventas_nulas.head(20)

shape: (20, 16)
┌────────────┬──────────┬────────────┬────────────┬───┬───────────┬───────────┬───────────┬────────┐
│ Name       ┆ Platform ┆ Year_of_Re ┆ Genre      ┆ … ┆ User_Scor ┆ User_Coun ┆ Developer ┆ Rating │
│ ---        ┆ ---      ┆ lease      ┆ ---        ┆   ┆ e         ┆ t         ┆ ---       ┆ ---    │
│ str        ┆ str      ┆ ---        ┆ str        ┆   ┆ ---       ┆ ---       ┆ str       ┆ str    │
│            ┆          ┆ f64        ┆            ┆   ┆ f64       ┆ f64       ┆           ┆        │
╞════════════╪══════════╪════════════╪════════════╪═══╪═══════════╪═══════════╪═══════════╪════════╡
│ Wii Sports ┆ Wii      ┆ 2006.0     ┆ Sports     ┆ … ┆ 8.0       ┆ 322.0     ┆ Nintendo  ┆ E      │
│ Super      ┆ NES      ┆ 1985.0     ┆ Platform   ┆ … ┆ null      ┆ null      ┆ null      ┆ null   │
│ Mario      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Bros.      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Mario Kart ┆ Wii      ┆ 2008.0     ┆ Racing     ┆ … ┆ 8.3       ┆ 709.0     ┆ Nintendo  ┆ E      │
│ Wii        ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Wii Sports ┆ Wii      ┆ 2009.0     ┆ Sports     ┆ … ┆ 8.0       ┆ 192.0     ┆ Nintendo  ┆ E      │
│ Resort     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Pokemon    ┆ GB       ┆ 1996.0     ┆ Role-Playi ┆ … ┆ null      ┆ null      ┆ null      ┆ null   │
│ Red/Pokemo ┆          ┆            ┆ ng         ┆   ┆           ┆           ┆           ┆        │
│ n Blue     ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ …          ┆ …        ┆ …          ┆ …          ┆ … ┆ …         ┆ …         ┆ …         ┆ …      │
│ Grand      ┆ PS2      ┆ 2004.0     ┆ Action     ┆ … ┆ 9.0       ┆ 1588.0    ┆ Rockstar  ┆ M      │
│ Theft      ┆          ┆            ┆            ┆   ┆           ┆           ┆ North     ┆        │
│ Auto: San  ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Andreas    ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Super      ┆ SNES     ┆ 1990.0     ┆ Platform   ┆ … ┆ null      ┆ null      ┆ null      ┆ null   │
│ Mario      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ World      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Brain Age: ┆ DS       ┆ 2005.0     ┆ Misc       ┆ … ┆ 7.9       ┆ 50.0      ┆ Nintendo  ┆ E      │
│ Train Your ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Brain in…  ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Pokemon    ┆ DS       ┆ 2006.0     ┆ Role-Playi ┆ … ┆ null      ┆ null      ┆ null      ┆ null   │
│ Diamond/Po ┆          ┆            ┆ ng         ┆   ┆           ┆           ┆           ┆        │
│ kemon      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Pearl      ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
│ Super      ┆ GB       ┆ 1989.0     ┆ Platform   ┆ … ┆ null      ┆ null      ┆ null      ┆ null   │
│ Mario Land ┆          ┆            ┆            ┆   ┆           ┆           ┆           ┆        │
└────────────┴──────────┴────────────┴────────────┴───┴───────────┴───────────┴───────────┴────────┘

In [11]:
duckdb.sql("""
    SELECT Genre AS GENERO,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(100 * SUM(NA_Sales) / SUM(SUM(NA_Sales)) OVER (),2) AS Por,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(100 * SUM(EU_Sales) / SUM(SUM(EU_Sales)) OVER (),2) AS Por,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(100 * SUM(JP_Sales) / SUM(SUM(JP_Sales)) OVER (),2) AS Por,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(100 * SUM(Other_Sales) / SUM(SUM(Other_Sales)) OVER (),2) AS Por,
           ROUND(SUM(Global_Sales), 2) AS Total,
           ROUND(100 * SUM(Global_Sales) / SUM(SUM(Global_Sales)) OVER (),2) AS Por
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 11 DESC;
""").show()


FloatProgress(value=0.0, layout=Layout(width='auto'), style=ProgressStyle(bar_color='black'))

┌──────────────┬───────────┬────────┬────────┬────────┬────────┬────────┬────────┬────────┬─────────┬────────┐
│    GENERO    │ N_America │  Por   │ Europa │  Por   │ Japon  │  Por   │ Otros  │  Por   │  Total  │  Por   │
│   varchar    │  double   │ double │ double │ double │ double │ double │ double │ double │ double  │ double │
├──────────────┼───────────┼────────┼────────┼────────┼────────┼────────┼────────┼────────┼─────────┼────────┤
│ Action       │    849.56 │  19.82 │ 494.86 │  20.95 │ 153.02 │  12.06 │ 176.52 │  22.74 │ 1673.53 │  19.26 │
│ Sports       │    674.04 │  15.73 │ 370.09 │  15.67 │ 134.27 │  10.58 │ 133.26 │  17.17 │  1311.4 │  15.09 │
│ Shooter      │     583.4 │  13.61 │ 313.29 │  13.26 │  38.04 │    3.0 │ 103.44 │  13.32 │ 1037.95 │  11.94 │
│ Role-Playing │    327.27 │   7.64 │ 186.78 │   7.91 │ 352.63 │  27.79 │  59.39 │   7.65 │  925.69 │  10.65 │
│ Platform     │    441.33 │   10.3 │ 198.81 │   8.42 │ 129.16 │  10.18 │   50.6 │   6.52 │  819.66 │   9.43 │
│

In [23]:
duckdb.sql("""
    SELECT ROUND(SUM(NA_Sales),2) AS N_AMERICA,
           ROUND(100 * SUM(NA_Sales) / SUM(Global_Sales),2) AS PORCENTAJE,
           ROUND(SUM(EU_Sales),2) AS EUROPA,
           ROUND(100 * SUM(EU_Sales) / SUM(Global_Sales),2) AS PORCENTAJE,
           ROUND(SUM(JP_Sales),2) AS JAPON,
           ROUND(100 * SUM(JP_Sales) / SUM(Global_Sales),2) AS PORCENTAJE,
           ROUND(SUM(Other_Sales),2) AS OTROS,
           ROUND(100 * SUM(Other_Sales) / SUM(Global_Sales),2) AS PORCENTAJE
    FROM sin_ventas_nulas;
""").show()

┌───────────┬────────────┬─────────┬────────────┬─────────┬────────────┬────────┬────────────┐
│ N_AMERICA │ PORCENTAJE │ EUROPA  │ PORCENTAJE │  JAPON  │ PORCENTAJE │ OTROS  │ PORCENTAJE │
│  double   │   double   │ double  │   double   │ double  │   double   │ double │   double   │
├───────────┼────────────┼─────────┼────────────┼─────────┼────────────┼────────┼────────────┤
│   4285.54 │      49.32 │ 2362.15 │      27.18 │ 1268.69 │       14.6 │ 776.34 │       8.93 │
└───────────┴────────────┴─────────┴────────────┴─────────┴────────────┴────────┴────────────┘



In [12]:
duckdb.sql("""
    SELECT Platform AS PLATAFORMA,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(100 * SUM(NA_Sales) / SUM(SUM(NA_Sales)) OVER (),2) AS Por,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(100 * SUM(EU_Sales) / SUM(SUM(EU_Sales)) OVER (),2) AS Por,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(100 * SUM(JP_Sales) / SUM(SUM(JP_Sales)) OVER (),2) AS Por,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(100 * SUM(Other_Sales) / SUM(SUM(Other_Sales)) OVER (),2) AS Por,
           ROUND(SUM(Global_Sales), 2) AS Total,
           ROUND(100 * SUM(Global_Sales) / SUM(SUM(Global_Sales)) OVER (),2) AS Por
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 11 DESC;
""").show()

┌────────────┬───────────┬────────┬────────┬────────┬────────┬────────┬────────┬────────┬─────────┬────────┐
│ PLATAFORMA │ N_America │  Por   │ Europa │  Por   │ Japon  │  Por   │ Otros  │  Por   │  Total  │  Por   │
│  varchar   │  double   │ double │ double │ double │ double │ double │ double │ double │ double  │ double │
├────────────┼───────────┼────────┼────────┼────────┼────────┼────────┼────────┼────────┼─────────┼────────┤
│ PS2        │    572.33 │  13.35 │ 331.92 │  14.05 │ 136.81 │  10.78 │  189.7 │  24.44 │ 1230.51 │  14.16 │
│ X360       │    578.71 │   13.5 │ 260.42 │  11.02 │  12.03 │   0.95 │  82.83 │  10.67 │  934.04 │  10.75 │
│ PS3        │    377.37 │   8.81 │ 314.77 │  13.33 │  77.33 │    6.1 │ 128.55 │  16.56 │  897.69 │  10.33 │
│ Wii        │    489.26 │  11.42 │ 259.33 │  10.98 │  68.94 │   5.43 │  78.76 │  10.15 │  896.09 │  10.31 │
│ DS         │    370.29 │   8.64 │ 184.51 │   7.81 │  170.4 │  13.43 │  58.71 │   7.56 │  783.58 │   9.02 │
│ PS         │     

Ahora con DuckDb podemos Generar la Consulta para obtener los datos que consideremos para nustras visualizaciones y automaticamente se guardan en un Dataframe de Pandas

In [27]:
import pandas as pd

In [29]:
duckdb.sql("""
    SELECT Genre AS GENERO,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(SUM(Global_Sales), 2) AS Total,
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 6 DESC;
""").show()

┌──────────────┬───────────┬────────┬────────┬────────┬─────────┐
│    GENERO    │ N_America │ Europa │ Japon  │ Otros  │  Total  │
│   varchar    │  double   │ double │ double │ double │ double  │
├──────────────┼───────────┼────────┼────────┼────────┼─────────┤
│ Action       │    849.56 │ 494.86 │ 153.02 │ 176.52 │ 1673.53 │
│ Sports       │    674.04 │ 370.09 │ 134.27 │ 133.26 │  1311.4 │
│ Shooter      │     583.4 │ 313.29 │  38.04 │ 103.44 │ 1037.95 │
│ Role-Playing │    327.27 │ 186.78 │ 352.63 │  59.39 │  925.69 │
│ Platform     │    441.33 │ 198.81 │ 129.16 │   50.6 │  819.66 │
│ Misc         │    385.31 │ 204.97 │ 105.76 │  71.41 │  767.15 │
│ Racing       │    347.52 │ 229.84 │  54.95 │  75.27 │   707.3 │
│ Fighting     │    220.49 │  98.75 │  86.55 │  36.14 │   441.6 │
│ Simulation   │    178.32 │ 112.07 │   60.8 │  30.65 │  381.73 │
│ Adventure    │     101.1 │  61.13 │  50.26 │  16.56 │  228.72 │
│ Puzzle       │    110.35 │  47.81 │  54.65 │  12.21 │  224.91 │
│ Strategy

In [30]:
df_pandas_genero = duckdb.sql("""
    SELECT Genre AS GENERO,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(SUM(Global_Sales), 2) AS Total,
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 6 DESC;
""").fetchdf()

In [31]:
duckdb.sql("""
    SELECT Platform AS PLATAFORMA,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(SUM(Global_Sales), 2) AS Total,
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 6 DESC;
""").show()

┌────────────┬───────────┬────────┬────────┬────────┬─────────┐
│ PLATAFORMA │ N_America │ Europa │ Japon  │ Otros  │  Total  │
│  varchar   │  double   │ double │ double │ double │ double  │
├────────────┼───────────┼────────┼────────┼────────┼─────────┤
│ PS2        │    572.33 │ 331.92 │ 136.81 │  189.7 │ 1230.51 │
│ X360       │    578.71 │ 260.42 │  12.03 │  82.83 │  934.04 │
│ PS3        │    377.37 │ 314.77 │  77.33 │ 128.55 │  897.69 │
│ Wii        │    489.26 │ 259.33 │  68.94 │  78.76 │  896.09 │
│ DS         │    370.29 │ 184.51 │  170.4 │  58.71 │  783.58 │
│ PS         │     325.0 │ 206.92 │ 135.26 │  40.17 │  706.81 │
│ GBA        │    185.32 │  74.39 │  47.02 │   8.58 │  315.02 │
│ PS4        │    107.85 │ 139.61 │  15.62 │  48.13 │  311.14 │
│ PSP        │    105.44 │  63.78 │  74.93 │  39.86 │  283.93 │
│ 3DS        │     82.81 │  61.03 │  99.81 │  13.42 │  257.01 │
│  ·         │        ·  │     ·  │    ·   │     ·  │     ·   │
│  ·         │        ·  │     ·  │    ·

In [32]:
df_pandas_plataforma = duckdb.sql("""
    SELECT Platform AS PLATAFORMA,
           ROUND(SUM(NA_Sales), 2) AS N_America,
           ROUND(SUM(EU_Sales), 2) AS Europa,
           ROUND(SUM(JP_Sales), 2) AS Japon,
           ROUND(SUM(Other_Sales), 2) AS Otros,
           ROUND(SUM(Global_Sales), 2) AS Total,
    FROM sin_ventas_nulas  
    GROUP BY 1
    ORDER BY 6 DESC;
""").fetchdf()

In [33]:
df_pandas_genero.to_csv('data/04_ventas_juegos_genero_consolas.csv', index=False)

In [34]:
df_pandas_plataforma.to_csv(
    'data/04_ventas_juegos_plataformas_consolas.csv', index=False
)